nomad/state/state_store.go

```diff
@@ -5333,6 +5333,99 @@ func (s *StateStore) BootstrapACLTokens(msgType structs.MessageType, index uint6
 	return txn.Commit()
 }
 
+// UpsertOneTimeToken is used to create or update a set of ACL
+// tokens. Validating that we're not upserting an already-expired token is
+// made the responsibility of the caller to facilitate testing.
+func (s *StateStore) UpsertOneTimeToken(msgType structs.MessageType, index uint64, token *structs.OneTimeToken) error {
+	txn := s.db.WriteTxnMsgT(msgType, index)
+	defer txn.Abort()
+
+	// we expect the RPC call to set the ExpiresAt but if it's somehow not
+	// set, set it now.
+	if token.ExpiresAt.IsZero() {
+		time.Now().Add(time.Minute * 10)
+	}
```

Comment:
Since we own all the calling code, this is really only enabling a future bug where the expiration stops being set; I think we should just return an error here, enforcing the invariant

Reply:
That's a really good point.

Reply:
Fixed in 87e0103